Makefile

```diff
@@ -24,10 +24,13 @@ test_setup:
 	g++ -std=c++11 -isystem ${GTEST_DIR}/include -I${GTEST_DIR} -pthread -c ${GTEST_DIR}/src/gtest-all.cc; \
 	ar -rv libgtest.a gtest-all.o;
 
-test: test_setup config_parser_test
+test: test_setup config_parser_test http_response_test server_config_parser_test
 
 config_parser_test: test_setup config_parser.cc config_parser_test.cc
-	g++ -std=c++11 -isystem ${GTEST_DIR}/include -pthread config_parser_test.cc config_parser.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a -o config_parser_test
+	g++ -std=c++11 -isystem ${GTEST_DIR}/include -pthread config_parser_test.cc config_parser.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a -o $@ 
```

Comment:
We can probably make a Bash variable to hold all the test flags together for a cleaner version of the Makefile. I can do it in my branch that adds the gcov flags as well.